In [233]:
import pandas as pd
import numpy as np 

In [234]:
df = pd.read_csv('dataset/Disease_processed_data.csv')
df.head()

Disease           Symptom_0           Symptom_1                Symptom_2  \
0    AIDS      muscle_wasting   patches_in_throat               high_fever   
1    AIDS   patches_in_throat          high_fever   extra_marital_contacts   
2    AIDS      muscle_wasting          high_fever   extra_marital_contacts   
3    AIDS      muscle_wasting   patches_in_throat   extra_marital_contacts   
4    AIDS      muscle_wasting   patches_in_throat               high_fever   

                 Symptom_3 Symptom_4 Symptom_precaution_0  \
0   extra_marital_contacts  headache      avoid open cuts   
1               high_fever  headache      avoid open cuts   
2               high_fever  headache      avoid open cuts   
3               high_fever  headache      avoid open cuts   
4               high_fever  headache      avoid open cuts   

   Symptom_precaution_1 Symptom_precaution_2 Symptom_precaution_3  \
0  wear ppe if possible       consult doctor            follow up   
1  wear ppe if possible       consult doctor            follow up   
2  wear ppe if possible       consult doctor            follow up   
3  wear ppe if possible       consult doctor            follow up   
4  wear ppe if possible       consult doctor            follow up   

                                 Symptom_Description  
0  Acquired immunodeficiency syndrome (AIDS) is a...  
1  Acquired immunodeficiency syndrome (AIDS) is a...  
2  Acquired immunodeficiency syndrome (AIDS) is a...  
3  Acquired immunodeficiency syndrome (AIDS) is a...  
4  Acquired immunodeficiency syndrome (AIDS) is a...

In [235]:
df.columns

Index(['Disease', 'Symptom_0', 'Symptom_1', 'Symptom_2', 'Symptom_3',
       'Symptom_4', 'Symptom_precaution_0', 'Symptom_precaution_1',
       'Symptom_precaution_2', 'Symptom_precaution_3', 'Symptom_Description'],
      dtype='object')

In [236]:
# df['Disease'] = df['Disease'].str.join(' ','_')
df['Disease'] = df['Disease'].str.strip()

In [237]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 313 entries, 0 to 312
Data columns (total 11 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Disease               313 non-null    object
 1   Symptom_0             313 non-null    object
 2   Symptom_1             313 non-null    object
 3   Symptom_2             313 non-null    object
 4   Symptom_3             313 non-null    object
 5   Symptom_4             313 non-null    object
 6   Symptom_precaution_0  313 non-null    object
 7   Symptom_precaution_1  313 non-null    object
 8   Symptom_precaution_2  313 non-null    object
 9   Symptom_precaution_3  313 non-null    object
 10  Symptom_Description   313 non-null    object
dtypes: object(11)
memory usage: 27.0+ KB


In [238]:
df['Disease'] = df['Disease'].str.replace(' ','_')

In [239]:
df1 = df[['Disease', 'Symptom_0', 'Symptom_1', 'Symptom_2', 'Symptom_3',
       'Symptom_4']]
df1.head()

Disease           Symptom_0           Symptom_1                Symptom_2  \
0    AIDS      muscle_wasting   patches_in_throat               high_fever   
1    AIDS   patches_in_throat          high_fever   extra_marital_contacts   
2    AIDS      muscle_wasting          high_fever   extra_marital_contacts   
3    AIDS      muscle_wasting   patches_in_throat   extra_marital_contacts   
4    AIDS      muscle_wasting   patches_in_throat               high_fever   

                 Symptom_3 Symptom_4  
0   extra_marital_contacts  headache  
1               high_fever  headache  
2               high_fever  headache  
3               high_fever  headache  
4               high_fever  headache

In [240]:
def fun1(columns):
    for i in columns:
        df1[i] = df1[i].str.strip()

fun1(['Disease', 'Symptom_0', 'Symptom_1', 'Symptom_2', 'Symptom_3',
       'Symptom_4'])

C:\Users\PAWAN\AppData\Local\Temp\ipykernel_16736\2205227840.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1[i] = df1[i].str.strip()


In [241]:
df1['Disease'] = df1['Disease'].str.replace(' ','_')

C:\Users\PAWAN\AppData\Local\Temp\ipykernel_16736\1794962573.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['Disease'] = df1['Disease'].str.replace(' ','_')


In [242]:
X = df1[['Symptom_0', 'Symptom_1', 'Symptom_2', 'Symptom_3',
       'Symptom_4']]
y = df1['Disease']

In [243]:
X = pd.get_dummies(X,columns=['Symptom_0', 'Symptom_1', 'Symptom_2', 'Symptom_3',
       'Symptom_4'])

In [244]:
X

Symptom_0_acidity  Symptom_0_back_pain  Symptom_0_bladder_discomfort  \
0                False                False                         False   
1                False                False                         False   
2                False                False                         False   
3                False                False                         False   
4                False                False                         False   
..                 ...                  ...                           ...   
308              False                False                         False   
309              False                False                         False   
310              False                False                         False   
311              False                False                         False   
312              False                False                         False   

     Symptom_0_breathlessness  Symptom_0_burning_micturition  \
0                       False                          False   
1                       False                          False   
2                       False                          False   
3                       False                          False   
4                       False                          False   
..                        ...                            ...   
308                     False                          False   
309                     False                          False   
310                     False                          False   
311                     False                          False   
312                     False                          False   

     Symptom_0_chest_pain  Symptom_0_chills  Symptom_0_constipation  \
0                   False             False                   False   
1                   False             False                   False   
2                   False             False                   False   
3                   False             False                   False   
4                   False             False                   False   
..                    ...               ...                     ...   
308                 False             False                   False   
309                 False             False                   False   
310                 False             False                   False   
311                 False             False                   False   
312                 False             False                   False   

     Symptom_0_continuous_sneezing  Symptom_0_cough  ...  \
0                            False            False  ...   
1                            False            False  ...   
2                            False            False  ...   
3                            False            False  ...   
4                            False            False  ...   
..                             ...              ...  ...   
308                          False            False  ...   
309                          False            False  ...   
310                          False            False  ...   
311                          False            False  ...   
312                          False            False  ...   

     Symptom_4_spotting_urination  Symptom_4_stiff_neck  Symptom_4_sweating  \
0                           False                 False               False   
1                           False                 False               False   
2                           False                 False               False   
3                           False                 False               False   
4                           False                 False               False   
..                            ...                   ...                 ...   
308                         False                 False               False   
309                         False                 False               False   
310                       

In [245]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.25,random_state=0)

In [246]:
print(X_train.shape)
print(X_test.shape)

(234, 234)
(79, 234)


In [247]:
def fun1(column):
    for i in column:
        X_train[i] = X_train[i].astype('int')

fun1(X_train.columns)


In [248]:
def fun2(column):
    for i in column:
        X_test[i] = X_test[i].astype('int')

fun2(X_test.columns)

In [249]:
X_train

Symptom_0_acidity  Symptom_0_back_pain  Symptom_0_bladder_discomfort  \
73                   0                    0                             0   
306                  0                    0                             0   
297                  0                    0                             0   
161                  0                    0                             0   
308                  0                    0                             0   
..                 ...                  ...                           ...   
251                  0                    0                             0   
192                  0                    0                             0   
117                  0                    0                             0   
47                   0                    0                             0   
172                  0                    0                             0   

     Symptom_0_breathlessness  Symptom_0_burning_micturition  \
73                          0                              0   
306                         0                              0   
297                         0                              0   
161                         0                              0   
308                         0                              0   
..                        ...                            ...   
251                         0                              0   
192                         0                              0   
117                         0                              0   
47                          0                              0   
172                         0                              0   

     Symptom_0_chest_pain  Symptom_0_chills  Symptom_0_constipation  \
73                      0                 0                       0   
306                     0                 0                       0   
297                     0                 1                       0   
161                     0                 0                       0   
308                     0                 0                       0   
..                    ...               ...                     ...   
251                     0                 0                       0   
192                     0                 0                       0   
117                     0                 0                       0   
47                      0                 0                       0   
172                     0                 0                       0   

     Symptom_0_continuous_sneezing  Symptom_0_cough  ...  \
73                               0                0  ...   
306                              0                0  ...   
297                              0                0  ...   
161                              0                0  ...   
308                              0                0  ...   
..                             ...              ...  ...   
251                              0                0  ...   
192                              0                0  ...   
117                              0                0  ...   
47                               0                0  ...   
172                              0                0  ...   

     Symptom_4_spotting_urination  Symptom_4_stiff_neck  Symptom_4_sweating  \
73                              0                     0                   0   
306                             0                     0                   0   
297                             0                     0                   0   
161                             0                     0                   0   
308                             0                     0                   0   
..                            ...                   ...                 ...   
251                             0                     0                   0   
192                             0                     0                   0   
117                       

In [250]:
X_test

Symptom_0_acidity  Symptom_0_back_pain  Symptom_0_bladder_discomfort  \
65                   0                    0                             0   
253                  0                    0                             0   
208                  0                    0                             0   
156                  0                    0                             0   
108                  0                    0                             0   
..                 ...                  ...                           ...   
188                  0                    0                             0   
103                  0                    0                             0   
159                  0                    0                             0   
90                   0                    0                             0   
129                  0                    0                             0   

     Symptom_0_breathlessness  Symptom_0_burning_micturition  \
65                          0                              0   
253                         0                              0   
208                         0                              0   
156                         0                              0   
108                         0                              0   
..                        ...                            ...   
188                         0                              0   
103                         0                              0   
159                         0                              0   
90                          0                              0   
129                         0                              0   

     Symptom_0_chest_pain  Symptom_0_chills  Symptom_0_constipation  \
65                      0                 0                       0   
253                     0                 0                       0   
208                     0                 0                       0   
156                     0                 0                       0   
108                     0                 0                       0   
..                    ...               ...                     ...   
188                     0                 0                       0   
103                     0                 0                       0   
159                     0                 0                       0   
90                      0                 0                       0   
129                     0                 0                       0   

     Symptom_0_continuous_sneezing  Symptom_0_cough  ...  \
65                               1                0  ...   
253                              0                0  ...   
208                              0                0  ...   
156                              0                0  ...   
108                              0                0  ...   
..                             ...              ...  ...   
188                              0                0  ...   
103                              0                0  ...   
159                              0                0  ...   
90                               0                0  ...   
129                              0                0  ...   

     Symptom_4_spotting_urination  Symptom_4_stiff_neck  Symptom_4_sweating  \
65                              0                     0                   0   
253                             0                     0                   0   
208                             0                     0                   0   
156                             0                     0                   0   
108                             1                     0                   0   
..                            ...                   ...                 ...   
188                             0                     0                   1   
103                             0                     0                   0   
159                       

Logistic Regression

In [251]:
from sklearn.linear_model import LogisticRegression

lr_classifier = LogisticRegression()
lr_classifier.fit(X_train,y_train)

LogisticRegression()

In [252]:
y_pred = lr_classifier.predict(X_test)

In [253]:
from sklearn import metrics
metrics.accuracy_score(y_pred,y_test)

0.8607594936708861

In [254]:
# X = pd.get_dummies(X,columns=['Symptom_0', 'Symptom_1', 'Symptom_2', 'Symptom_3',
#        'Symptom_4'])

In [255]:
# col = np.array(['muscle_wasting', 'high_fever', 'extra_marital_contacts','high_fever','headache'])
# # col = col.reshape(1,-1)
# col = pd.get_dummies(X,col)
# col


In [256]:
X_train

Symptom_0_acidity  Symptom_0_back_pain  Symptom_0_bladder_discomfort  \
73                   0                    0                             0   
306                  0                    0                             0   
297                  0                    0                             0   
161                  0                    0                             0   
308                  0                    0                             0   
..                 ...                  ...                           ...   
251                  0                    0                             0   
192                  0                    0                             0   
117                  0                    0                             0   
47                   0                    0                             0   
172                  0                    0                             0   

     Symptom_0_breathlessness  Symptom_0_burning_micturition  \
73                          0                              0   
306                         0                              0   
297                         0                              0   
161                         0                              0   
308                         0                              0   
..                        ...                            ...   
251                         0                              0   
192                         0                              0   
117                         0                              0   
47                          0                              0   
172                         0                              0   

     Symptom_0_chest_pain  Symptom_0_chills  Symptom_0_constipation  \
73                      0                 0                       0   
306                     0                 0                       0   
297                     0                 1                       0   
161                     0                 0                       0   
308                     0                 0                       0   
..                    ...               ...                     ...   
251                     0                 0                       0   
192                     0                 0                       0   
117                     0                 0                       0   
47                      0                 0                       0   
172                     0                 0                       0   

     Symptom_0_continuous_sneezing  Symptom_0_cough  ...  \
73                               0                0  ...   
306                              0                0  ...   
297                              0                0  ...   
161                              0                0  ...   
308                              0                0  ...   
..                             ...              ...  ...   
251                              0                0  ...   
192                              0                0  ...   
117                              0                0  ...   
47                               0                0  ...   
172                              0                0  ...   

     Symptom_4_spotting_urination  Symptom_4_stiff_neck  Symptom_4_sweating  \
73                              0                     0                   0   
306                             0                     0                   0   
297                             0                     0                   0   
161                             0                     0                   0   
308                             0                     0                   0   
..                            ...                   ...                 ...   
251                             0                     0                   0   
192                             0                     0                   0   
117                       

In [257]:
y_train

73                            Covid
306                  Varicose_veins
297                         Typhoid
161                     Hepatitis_D
308                  Varicose_veins
                   ...             
251    Paralysis_(brain_hemorrhage)
192                    Hypoglycemia
117                            GERD
47                      Chicken_pox
172                     Hepatitis_E
Name: Disease, Length: 234, dtype: object

Naive Baiyes

In [258]:
from sklearn.naive_bayes import GaussianNB
nb_classifier = GaussianNB()
nb_classifier.fit(X_train,y_train)

GaussianNB()

In [259]:
y_pred = nb_classifier.predict(X_test)

In [260]:
from sklearn.metrics import accuracy_score
metrics.accuracy_score(y_pred,y_test)

0.8734177215189873

Decision Tree

In [261]:
from sklearn.tree import DecisionTreeClassifier
dt_classifier = DecisionTreeClassifier()
dt_classifier.fit(X_train,y_train)

DecisionTreeClassifier()

In [262]:
y_pred = dt_classifier.predict(X_test)

In [263]:
metrics.accuracy_score(y_test,y_pred)

0.6835443037974683

from these models we see that the naive bayes algorithm giving us the 87.34 % accuracy but the decision tree algorithm model gives us the 69.62 % accuracy , so we choose naive baise algorithm instead of using the decision tree algorithm .

In [264]:
from pickle import dump
dump(nb_classifier,open('Models/nb_classifier_disease.pkl','wb'))

Testing

In [265]:
col = np.array(['muscle_wasting', 'high_fever', 'extra_marital_contacts','high_fever','headache'])
col = col.astype('object')
col

array(['muscle_wasting', 'high_fever', 'extra_marital_contacts',
       'high_fever', 'headache'], dtype=object)

In [266]:
# X = pd.get_dummies(X,columns=['Symptom_0', 'Symptom_1', 'Symptom_2', 'Symptom_3',
#        'Symptom_4'])

In [267]:
query_point = col
query_point = query_point.reshape(1,-1)
query_point = pd.DataFrame(query_point,columns=['Symptom_0', 'Symptom_1', 'Symptom_2', 'Symptom_3',
       'Symptom_4'])
query_point

Symptom_0   Symptom_1               Symptom_2   Symptom_3 Symptom_4
0  muscle_wasting  high_fever  extra_marital_contacts  high_fever  headache

In [268]:
point = pd.get_dummies(query_point,columns=['Symptom_0', 'Symptom_1', 'Symptom_2', 'Symptom_3',
       'Symptom_4'])
point = np.array(point)
point

array([[ True,  True,  True,  True,  True]])

In [269]:
# dt_classifier.predict(point)

In [270]:
quey  = np.random.randint(0,2,size=(234,))
quey = quey.reshape(1,-1)
quey

array([[1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 1, 0,
        1, 1, 1, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 1,
        1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0,
        0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0,
        1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0,
        0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0,
        1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0, 1, 0, 1, 1, 0, 0,
        0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 1, 0, 0, 1, 1, 1,
        1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 1, 1, 0,
        0, 1, 1, 1, 0, 1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0,
        0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0]])

In [271]:
dt_classifier.predict(quey)

C:\Users\PAWAN\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\sklearn\base.py:464: UserWarning: X does not have valid feature names, but DecisionTreeClassifier was fitted with feature names
  warnings.warn(


array(['Typhoid'], dtype=object)

In [272]:
nb_classifier.predict(quey)

C:\Users\PAWAN\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\sklearn\base.py:464: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(


array(['Covid'], dtype='<U29')

In [273]:
col = X_train.columns

In [274]:
dump(col,open('dataset/columns.pkl','wb'))

here is the way to production the model

In [275]:
cols = X_train.columns
symptoms = {}
for i in cols:
    symptoms[i] = 0
symptoms 

{'Symptom_0_acidity': 0,
 'Symptom_0_back_pain': 0,
 'Symptom_0_bladder_discomfort': 0,
 'Symptom_0_breathlessness': 0,
 'Symptom_0_burning_micturition': 0,
 'Symptom_0_chest_pain': 0,
 'Symptom_0_chills': 0,
 'Symptom_0_constipation': 0,
 'Symptom_0_continuous_sneezing': 0,
 'Symptom_0_cough': 0,
 'Symptom_0_cramps': 0,
 'Symptom_0_fatigue': 0,
 'Symptom_0_headache': 0,
 'Symptom_0_high_fever': 0,
 'Symptom_0_indigestion': 0,
 'Symptom_0_itching': 0,
 'Symptom_0_joint_pain': 0,
 'Symptom_0_loss_of_smell': 0,
 'Symptom_0_loss_of_taste': 0,
 'Symptom_0_mood_swings': 0,
 'Symptom_0_muscle_wasting': 0,
 'Symptom_0_muscle_weakness': 0,
 'Symptom_0_neck_pain': 0,
 'Symptom_0_pain_during_bowel_movements': 0,
 'Symptom_0_patches_in_throat': 0,
 'Symptom_0_pus_filled_pimples': 0,
 'Symptom_0_shivering': 0,
 'Symptom_0_skin_rash': 0,
 'Symptom_0_stiff_neck': 0,
 'Symptom_0_stomach_pain': 0,
 'Symptom_0_sunken_eyes': 0,
 'Symptom_0_tiredness': 0,
 'Symptom_0_vomiting': 0,
 'Symptom_0_weakness_in

In [276]:
len(symptoms)

234

In [277]:
symptoms.popitem()
query = np.array(['Symptom_0_loss_of_smell','Symptom_1_fatigue','Symptom_2_unsteadiness','Symptom_3_yellowish_skin'])
for i in query:
    symptoms[i] = 1

l = np.array(list(symptoms.values())).reshape(1,-1)
l


array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]])

In [278]:
nb_classifier.predict(l)

C:\Users\PAWAN\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\sklearn\base.py:464: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(


array(['Hepatitis_E'], dtype='<U29')

In [279]:
dt_classifier.predict(l)

C:\Users\PAWAN\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\sklearn\base.py:464: UserWarning: X does not have valid feature names, but DecisionTreeClassifier was fitted with feature names
  warnings.warn(


array(['Pneumonia'], dtype=object)

In [280]:
disease = nb_classifier.predict(l)

C:\Users\PAWAN\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\sklearn\base.py:464: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(


this is for finding the precautions

In [281]:
df = pd.read_csv('dataset2/disease_precaution.csv')
df.head()

Disease      Symptom_precaution_0      Symptom_precaution_1  \
0                 AIDS           avoid open cuts      wear ppe if possible   
1                 Acne                bath twice    avoid fatty spicy food   
2  Alcoholic hepatitis  stop alcohol consumption            consult doctor   
3              Allergy            apply calamine   cover area with bandage   
4            Arthritis                  exercise  use hot and cold therapy   

    Symptom_precaution_2         Symptom_precaution_3  
0         consult doctor                    follow up  
1  drink plenty of water      avoid too many products  
2             medication                    follow up  
3                    NaN  use ice to compress itching  
4        try acupuncture                      massage

In [282]:
# def fun3(cols):
#     df[i] = df[i].str.strip()

# fun3(df.columns)
# # fun1(df.columns)

df['Disease'] = df['Disease'].str.strip()
df['Disease'] = df['Disease'].str.replace(' ','_')

In [283]:
dump(df,open('dataset/app_precaution_data.pkl','wb'))

In [284]:
df2 = df[df['Disease']==disease[0]]
df2

Disease      Symptom_precaution_0 Symptom_precaution_1  \
23  Hepatitis_E  stop alcohol consumption                 rest   

   Symptom_precaution_2 Symptom_precaution_3  
23       consult doctor           medication

In [285]:
precuation = np.array(df2).reshape(1,-1)
precuation = precuation[0][1:]
precuation

array(['stop alcohol consumption', 'rest', 'consult doctor', 'medication'],
      dtype=object)

this is used to find the description of the disease

In [286]:
df3 = pd.read_csv('dataset2/disease_description.csv')
df3.head()

Disease                                Symptom_Description
0                 AIDS  Acquired immunodeficiency syndrome (AIDS) is a...
1                 Acne  Acne vulgaris is the formation of comedones, p...
2  Alcoholic hepatitis  Alcoholic hepatitis is a diseased, inflammator...
3              Allergy  An allergy is an immune system response to a f...
4            Arthritis  Arthritis is the swelling and tenderness of on...

In [287]:
df3['Disease'] = df3['Disease'].str.strip()
df3['Disease'] = df3['Disease'].str.replace(' ','_')

In [288]:
dump(df3,open('dataset/app_description_data.pkl','wb'))

In [289]:
description = df3[df3['Disease']==disease[0]]
description

Disease                                Symptom_Description
23  Hepatitis_E  A rare form of liver inflammation caused by in...

In [290]:
description = np.array(description)
description

array([['Hepatitis_E',
        'A rare form of liver inflammation caused by infection with the hepatitis E virus (HEV). It is transmitted via food or drinks handled by an infected person or through infected water supplies in areas where the faecal matter may get into the water. Hepatitis E does not cause chronic liver disease.']],
      dtype=object)

In [291]:
description = description[0][1]
description

'A rare form of liver inflammation caused by infection with the hepatitis E virus (HEV). It is transmitted via food or drinks handled by an infected person or through infected water supplies in areas where the faecal matter may get into the water. Hepatitis E does not cause chronic liver disease.'